packages/node/src/integrations/contextlines.ts

```diff
@@ -47,42 +132,64 @@ export const contextLinesIntegration = defineIntegration(_contextLinesIntegratio
 async function addSourceContext(event: Event, contextLines: number): Promise<Event> {
   // keep a lookup map of which files we've already enqueued to read,
   // so we don't enqueue the same file multiple times which would cause multiple i/o reads
-  const enqueuedReadSourceFileTasks: Record<string, number> = {};
-  const readSourceFileTasks: Promise<string[] | null>[] = [];
+  const filesToLines: Record<string, number[]> = {};
 
   if (contextLines > 0 && event.exception?.values) {
     for (const exception of event.exception.values) {
-      if (!exception.stacktrace?.frames) {
+      if (!exception.stacktrace?.frames?.length) {
         continue;
       }
 
-      // We want to iterate in reverse order as calling cache.get will bump the file in our LRU cache.
-      // This ends up prioritizes source context for frames at the top of the stack instead of the bottom.
+      // Maps preserve insertion order, so we iterate in reverse, starting at the
+      // outermost frame and closer to where the exception has occurred (poor mans priority)
       for (let i = exception.stacktrace.frames.length - 1; i >= 0; i--) {
         const frame = exception.stacktrace.frames[i];
-        // Call cache.get to bump the file to the top of the cache and ensure we have not already
-        // enqueued a read operation for this filename
-        if (frame.filename && !enqueuedReadSourceFileTasks[frame.filename] && !FILE_CONTENT_CACHE.get(frame.filename)) {
-          readSourceFileTasks.push(_readSourceFile(frame.filename));
-          enqueuedReadSourceFileTasks[frame.filename] = 1;
+
+        // Cases where collecting context lines is either not useful or possible
+        // - .min.js files are and not useful since they dont point to the original source
+        // - node: prefixed modules are part of the runtime and cannot be resolved to a file
```

Comment:
As per https://nodejs.org/api/esm.html#import-specifiers we want to skip `data:` as well.